Library/src/lab1/Library.java

```diff
@@ -1,11 +1,78 @@
 package lab1;
 
-/**
- * A representation of a library of books.
- * @author <a href="mailto:rkhatchadourian@citytech.cuny.edu">Raffi Khatchadourian</a>.
- */
 public class Library {
-    // TODO: Add the missing implementation to this class
+
+    String address;
+    Book[] collection = {};
+
+    public Library(String location) {
+        address = location;
+    }
+
+    public void addBook(Book book) {
+        Book[] add = new Book[collection.length + 1];
+        for (int i = 0; i < collection.length; i++) {
+            add[i] = collection[i];
+        }
+        add[collection.length] = book;
+        collection = add;
```

Comment:
This is too complicated. You can use a partially filled array instead. You can set a maximum number of books.